### PR TITLE
feat: arrow and some raycast changes.

### PR DESCRIPTION
**Pull request by Tebarem**
We finally have arrows.

### CODE REVIEW

File: crates/hyperion/src/simulation/mod.rs [hunk 1]
```diff
@@ -717,6 +762,62 @@ impl Module for SimModule {
...
+
+                    debug!("collision = {collision:?}");
+
+                    velocity.velocity = Vec3::ZERO;
+
+                    // Set arrow position to the collision location
+                    **position = collision.normal;
+
+                    blocks
+                        .set_block(collision.location, BlockState::DIRT)
```
Comment (CuzImClicks) [inline crates/hyperion/src/simulation/mod.rs:816]: used for debugging?

Comment (Tebarem) [reply crates/hyperion/src/simulation/mod.rs:816]: sure but its used to show that you can do whatever

Comment (Tebarem) [reply crates/hyperion/src/simulation/mod.rs:816]: line 813 is to set the arrow on the block it hit

**Comment by andrewgazelka** [comment]
ok I made some changes @Tebarem 

https://github.com/andrewgazelka/hyperion/pull/658/files/df7890edb1c7ed8139755805bde63bc93cf1ab93..58a27491ad1ae82bf3a2c1129d10af724e1510ea

I am approving for now if you want to merge.

**Review by andrewgazelka — APPROVED** [review]

**Comment by andrewgazelka** [comment]
merging in; if there are issues feel free to resolve in future PR

**Pull request merged**